src/datasources/accounts/accounts.datasource.ts

```diff
@@ -192,4 +210,38 @@ export class AccountsDatasource implements IAccountsDatasource, OnModuleInit {
       );
     }
   }
+
+  /**
+   * Checks if the client IP address has reached the account creation rate limit.
+   *
+   * NOTE: the rate limit is implemented in the datasource layer for this use case
+   * because we need to restrict the actual creation of accounts, not merely
+   * the attempts to create them.
+   *
+   * If the client IP address is invalid, a warning is logged.
+   * If the client IP address is valid and rate limit is reached, a {@link LimitReachedError} is thrown.
+   *
+   * @param clientIp - client IP address.
+   */
+  private async checkCreationRateLimit(clientIp: string): Promise<void> {
+    const { success: isValidIp } = IpSchema.safeParse(clientIp);
+    if (!clientIp || !isValidIp) {
+      this.loggingService.warn(
+        `Invalid client IP while creating account: ${clientIp}`,
+      );
+    } else {
+      const current = await this.cacheService.increment(
+        CacheRouter.getRateLimitCacheKey(
+          `${AccountsDatasource.ACCOUNT_CREATION_CACHE_PREFIX}_${clientIp}`,
+        ),
+        this.accountCreationRateLimitPeriodSeconds,
+      );
+      if (current > this.accountCreationRateLimitCalls) {
+        this.loggingService.warn(
+          `Limit of ${this.accountCreationRateLimitCalls} reached for IP ${clientIp}`,
+        );
+        throw new LimitReachedError();
```

Comment:
Which http code would a client get?
this Error is not extending HttpException. Is it necessary to implement an ExceptionFilter for this?

Reply:
With this implementation, the client would get a generic 500 status code error.

I've created a PR to extend `HttpException` and set the correct status codes for both accounts and Counterfactual Safes rate limit hit errors on creation: https://github.com/safe-global/safe-client-gateway/pull/1865

Thank you!